astropy/io/fits/tests/test_image.py

```diff
@@ -73,7 +73,7 @@ def test_constructor_copies_header(self):
         # Original header should be unchanged
         assert phdr['FILENAME'] == 'labq01i3q_rawtag.fits'
 
-    @raises(ValueError)
+    @raises(IndexError)
```

Comment:
@embray - I implemented my last suggestion (https://github.com/astropy/astropy/pull/6082/commits/7381a700dad51f50724f7ca995491edddb2ac56b), this way `_read_all` cannot be set to True after the file is closed. Does this work for you ?